File: InitFunc/InitFunc/Person.m

//
//  Person.m
//  InitFunc
//
//  Created by mfang032 on 3/2/16.
//  Copyright © 2016 mfang032. All rights reserved.
//

#import "Person.h"

@implementation Person
- (instancetype)init {
    if (self = [super init]) {
        _name = @"";
    }
    return self;
}

+ (instancetype)person {
    return [self init];
}
@end

File: 多线程/ThreadSafe/ThreadSafe/Person.swift
//
//  Person.swift
//  ThreadSafe
//
//  Created by mfang032 on 2017/4/6.
//  Copyright © 2017年 mfang032. All rights reserved.
//

import UIKit

class Person: NSObject {
    
    var name: String?
    var age: Int?
    
    init(_ name: String, _ age: Int) {
        self.name = name
        self.age = age
    }
    
    func update(name: String, age: Int) {
        self.name = name
        self.age = age
    }
    
    override var description: String {
        return "name: \(String(describing: name ?? "")), age: \(String(describing: age ?? 0))"
    }
    
}
